src/plugins/validation/oas3/semantic-validators/responses.js

```diff
@@ -84,7 +93,49 @@ module.exports.validate = function({ resolvedSpec }, config) {
   return { errors: result.error, warnings: result.warning };
 };
 
-function isResponseCode(code) {
+function getResponseCodes(responseObj) {
+  const allResponseCodes = Object.keys(responseObj).filter(
+    code => isStatusCode(code) || code === 'default'
+  );
+  const statusCodes = allResponseCodes.filter(code => isStatusCode(code));
+  const successCodes = statusCodes.filter(code => code.slice(0, 1) === '2');
+  return [allResponseCodes, statusCodes, successCodes];
+}
+
+function isStatusCode(code) {
   const allowedFirstDigits = ['1', '2', '3', '4', '5'];
   return code.length === 3 && allowedFirstDigits.includes(code.slice(0, 1));
 }
+
+function validateResponseBody(
+  responseObj,
+  allResponseCodes,
+  result,
+  path,
+  binaryStringStatus
+) {
+  for (const responseCode of allResponseCodes) {
+    const responseBodyContent = responseObj[responseCode].content;
+    if (responseBodyContent) {
+      Object.keys(responseBodyContent).forEach(function(mimeType) {
+        if (mimeType === 'application/json') {
+          const schemaPath = `${path.join(
+            '.'
+          )}.${responseCode}.content.${mimeType}.schema`;
+          const octetSequencePaths = octetSequences(
+            responseBodyContent[mimeType].schema,
+            schemaPath
+          );
+          const message =
+            'No well-defined way to represent bare octets (type: string, format: binary) as JSON.';
+          for (const p of octetSequencePaths) {
+            result[binaryStringStatus].push({
+              path: p.split('.'),
+              message: message
```

Comment:
There seems to be contention in how to represent the path here.  We need to join it to call `octetSequencePaths`, but then split it again to set result.  Can't we just use one format consistently?

Reply:
Some of the semantic validators use arrays and others use strings with `.` as the delimiter. For this PR I just matched the convention of each validator, but I would like to standardize the paths to use arrays in the next PR.